lib/shared/models/flash-state.js

```diff
@@ -114,25 +114,25 @@ exports.unsetFlashingFlag = (results) => {
  * });
  */
 exports.setProgressState = (state) => {
+  const pickedState = _.pick(state, [
+    'type',
+    'percentage',
+    'eta',
+    'speed'
+  ])
+
+  const humanSpeed = units.bytesToClosestUnit(state.speed, '/s')
+
+  const data = _.assign({}, pickedState, {
+    humanSpeed,
+    percentage: _.isFinite(state.percentage)
+      ? Math.floor(state.percentage)
+      : state.percentage
+  })
```

Comment:
I think this is overcomplicating things heavily for no apparent reason. Why not just do the following?

```js
state.humanSpeed = units.bytesToClosestUnit(state.speed, '/s')

store.dispatch({
  type: store.Actions.SET_FLASH_STATE,
  data: state
})
```

Reply:
I'd assume that @Shou is trying to preserve the "const-ness" of `state` by creating a new object rather than modifying `state` in place? \*shrug\*

BTW do any of the unit-tests check what happens when `_.isFinite(state.percentage)` returns false?

And just to be pedantic I guess there's no need for `percentage` in `pickedState` because `data.percentage` is being set directly from `state.percentage` ;)